src/npfctl/npf_build.c

```diff
@@ -912,21 +930,29 @@ npfctl_fill_table(nl_table_t *tl, u_int type, const char *fname)
  * if required, fill with contents from a file.
  */
 void
-npfctl_build_table(const char *tname, u_int type, const char *fname)
+npfctl_build_table(const char *tname, int tid, u_int type, const char *fname)
 {
 	nl_table_t *tl;
 
-	tl = npf_table_create(tname, npfctl_tid_counter++, type);
+	if (tid < 0) {
+		tid = npfctl_tid_counter++;
+	}
+
+	tl = npf_table_create(tname, tid, type);
 	assert(tl != NULL);
 
 	if (fname) {
 		npfctl_fill_table(tl, type, fname);
 	} else if (type == NPF_TABLE_CONST) {
-		yyerror("table type 'const' must be loaded from a file");
```

Comment:
- `yyerror()` shows the location in the config where this error happen, so we need to preserve this.
- It might (or might not) be easier to just create a separate version of this function..

Reply:
OK, I'll rethink this a little. In an earlier iteration I had split the table creation and population code from `npfctl_build_table()` into a `npfctl_load_table()` function, which created and populated the table and returned it.  `npf_build_table()` simply called it and did the other bits required by the parser (allocating the next available `tid` and inserting the table to the config). Perhaps I should go to back to this design and also ensure `NPF_TABLE_CONST` have a filename only in the parser version?

Another thing to note is that currently the `npfctl_table_replace()` command parser always passes a value for `fname` to `npfctl_build_table()`; so that particular `else if` will never be evaluated in this case. In theory, this means the `yyerror()` could stay, but doing so would be a very unclear logic sequence to follow. I think splitting the function is probably a better idea.

In any case, there's currently nothing stopping the creation of passing `/dev/null` as filename to create an empty `NPF_TABLE_CONST`, and perhaps this check isn't so critical given that this PR allows us to replace it with a new table later?

Reply:
It can be optional in the table raplacement logic. However, in the configuration file, I think, semantically, the file name requirement should be enforced. Even if it's an empty file, the user can populate in later and just do `npfctl reload`, rather than mess with the config.